### PR TITLE
Data aware scheduling docs edits

### PR DESCRIPTION
**Pull request by lzdanski**
---

Copy edits for 2.9 Data Aware Scheduling feature docs.

### CODE REVIEW

**Comment by lzdanski** [comment]
@sunank200 -  Docs copyediting for https://github.com/apache/airflow/pull/37101. I took a look at the PR's you shared with me, https://github.com/apache/airflow/pull/37771 and https://github.com/apache/airflow/pull/38576, and I think copyedits for this content might be covered by https://github.com/apache/airflow/pull/38505.

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst
Comment (uranusjr) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst]:
```suggestion
An Airflow dataset is a logical grouping of data. Upstream producer tasks can update datasets, and dataset updates contribute to scheduling downstream consumer DAGs.
```

This should be uniform. (Can be all capitalized instead but you don’t in all other cases.)

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst
Comment (uranusjr) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst:71]: We should probably add a link to the Wikipedia entry on URI somewhere too. https://en.wikipedia.org/wiki/Uniform_Resource_Identifier

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst
Comment (lzdanski) [reply docs/apache-airflow/authoring-and-scheduling/datasets.rst]: switched to lowercase - we have it capitalized when it's used in code examples, seemed like a good way to make a quick distinction.

**Comment by lzdanski** [comment]
Wait on review from @sunank200 before merging!

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst
Comment (potiuk) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst:26]: Since it's waiting for @sunank200 - comment here to prevent accidental merge :)

Comment (sunank200) [reply docs/apache-airflow/authoring-and-scheduling/datasets.rst:26]: LGTM

**Review by sunank200 — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**
